footbot/optimiser/team_selector.py

```diff
@@ -90,6 +93,7 @@ def select_team(
         # player number constraints
         np.ones(len(players)) @ first_team == 11,
         np.ones(len(players)) @ captain == 1,
+        np.ones(len(players)) @ vice == 1,
```

Comment:
I think we need two more constraints for edge cases:
```
# first team contains vice
first_team - vice >= np.zeros(len(players)),
# captain and vice are different
captain + vice <= np.ones(len(players)),

```

Reply:
ok. curious that it doesnt pick captain = vice when i test it though? also presumably there are circumstances in which VC on bench is good choice?

Reply:
I'm not sure why it doesn't pick captain and vice as the same without that constraint. Seems like it would be the optimal thing to do, as without that constraint it's sort of equivalent to just increasing the captain factor.

Reply:
I don't think it would ever be optimal to have your vice (i.e. second most promising player) on the bench. Strictly the constraint to have the vice (and the captain) in the first team isn't necessary, as those would be suboptimal choices given the low bench factor. But I don't think it hurts to enforce it. What do you think?

Reply:
hypothetical scenario: two goalies have best prediction, you'd want to captain the first team one and VC the other no? if we think the optimiser is making suboptimal choices we should try to fix that rather than add additional constraints imo

Reply:
Yeah - that makes sense not to constrain the captain or vice to be in the first team. You've convinced me!

I still think we should constrain the captain and vice not to be the same player though. Thoughts?

Reply:
yep agreed. once API is back up i will experiment a bit with factor to try to see why it isn't already doing this

Reply:
i added `captain + vice <= np.ones(len(players)),` and now it _always_ picks same player as cap and vice

Reply:
lol

Reply:
curiouser and curiouser. I'll try debug tomorrow if I have time.

Reply:
Okay - I've fixed it:

https://github.com/thomas-woodruff/footbot/commit/f1d023cc94d901a8f726c9826c645c29f1a1b8dc

Not sure what the procedure is now. Should I merge in that branch? Or merge that commit into this branch somehow?

Reply:
good job yes merge that